### PR TITLE
New Resources: `azurerm_app_service_slot` & `azurerm_app_service_active_slot`

### PR DESCRIPTION
**Pull request by jeffreyCline**
This PR adds support for App Service Slot and Active Slot features to the Azure Terraform Provider.

**Added Resources**
* `resource_arm_app_service_active_slot`
* `resource_arm_app_service_active_slot_test`
* `resource_arm_app_service_slot`
* `resource_arm_app_service_slot_test`
* `app_service_slot.html.markdown`
* `app_service_active_slot.html.markdown`

**Updated Resources**
* `azurerm.erb`
* `provider`

[Test Results](https://gist.github.com/jeffreyCline/34f239d6c2f76128ffdc9da164a15d9c)

### CODE REVIEW

**Comment by metacpp** [comment]
@tombuildsstuff this PR has been verified by @jeffreyCline in our own environment. See test results link to Gist.

File: azurerm/resource_arm_app_service_slot_test.go [hunk 1]
```diff
@@ -0,0 +1,1269 @@
...
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  app_service_plan_id = "${azurerm_app_service_plan.test.id}"
+}
+
+resource "azurerm_app_service_slot" "test" {
+  name                = "acctestASSlot-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+	app_service_plan_id = "${azurerm_app_service_plan.test.id}"
+	app_service_name    = "${azurerm_app_service.test.name}"
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 2]
```diff
@@ -0,0 +1,1269 @@
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 3]
```diff
@@ -0,0 +1,1269 @@
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 4]
```diff
@@ -0,0 +1,1269 @@
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 5]
```diff
@@ -0,0 +1,1269 @@
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 6]
```diff
@@ -0,0 +1,1269 @@
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 7]
```diff
@@ -0,0 +1,1269 @@
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 8]
```diff
@@ -0,0 +1,1269 @@
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 9]
```diff
@@ -0,0 +1,1269 @@
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 10]
```diff
@@ -0,0 +1,1269 @@
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 11]
```diff
@@ -0,0 +1,1269 @@
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 12]
```diff
@@ -0,0 +1,1269 @@
...
+
+  site_config {
+  	default_documents = [
+		"first.html",
+		"second.jsp",
+		"third.aspx",
+  	]
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 13]
```diff
@@ -0,0 +1,1269 @@
...
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  app_service_plan_id = "${azurerm_app_service_plan.test.id}"
+}
+
+resource "azurerm_app_service_slot" "test" {
+  name                = "acctestASSlot-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+	app_service_plan_id = "${azurerm_app_service_plan.test.id}"
+	app_service_name    = "${azurerm_app_service.test.name}"
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 14]
```diff
@@ -0,0 +1,1269 @@
...
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  app_service_plan_id = "${azurerm_app_service_plan.test.id}"
+}
+
+resource "azurerm_app_service_slot" "test" {
+  name                    = "acctestASSlot-%d"
+  location                = "${azurerm_resource_group.test.location}"
+  resource_group_name     = "${azurerm_resource_group.test.name}"
+	app_service_plan_id     = "${azurerm_app_service_plan.test.id}"
+	app_service_name        = "${azurerm_app_service.test.name}"
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 15]
```diff
@@ -0,0 +1,1269 @@
...
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  app_service_plan_id = "${azurerm_app_service_plan.test.id}"
+}
+
+resource "azurerm_app_service_slot" "test" {
+  name                = "acctestASSlot-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+	app_service_plan_id = "${azurerm_app_service_plan.test.id}"
+	app_service_name    = "${azurerm_app_service.test.name}"
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 16]
```diff
@@ -0,0 +1,1269 @@
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 17]
```diff
@@ -0,0 +1,1269 @@
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_slot_test.go [hunk 18]
```diff
@@ -0,0 +1,1269 @@
...
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  app_service_plan_id = "${azurerm_app_service_plan.test.id}"
+}
+
+resource "azurerm_app_service_slot" "test" {
+	name                = "acctestASSlot-%d"
+	location            = "${azurerm_resource_group.test.location}"
+	resource_group_name = "${azurerm_resource_group.test.name}"
+	app_service_plan_id = "${azurerm_app_service_plan.test.id}"
+	app_service_name    = "${azurerm_app_service.test.name}"
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_active_slot_test.go [hunk 19]
```diff
@@ -0,0 +1,200 @@
...
+	
+	resource "azurerm_app_service_plan" "test" {
+		name                = "acctestASP-%d"
+		location            = "${azurerm_resource_group.test.location}"
+		resource_group_name = "${azurerm_resource_group.test.name}"
+	
+		sku {
+		tier = "Standard"
+		size = "S1"
+		}
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_active_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_active_slot_test.go [hunk 20]
```diff
@@ -0,0 +1,200 @@
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_active_slot_test.go]: _minor_ could we sort out the spacing here?

File: azurerm/resource_arm_app_service_active_slot_test.go [hunk 21]
```diff
@@ -0,0 +1,200 @@
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service_active_slot_test.go]: _minor_ could we sort out the spacing here?

File: website/docs/r/app_service_active_slot.html.markdown [hunk 22]
```diff
@@ -0,0 +1,55 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `resource_group_name` - (Required) The name of the resource group in which to create the App Service Slot component.
+
+* `app_service_name` - (Required) The name of the App Service within which to create the App Service Slot.  Changing this forces a new resource to be created.
+
+* `preserve_vnet` - (Required) `true` to preserve Virtual Network to the slot during swap; otherwise, `false`.
```
[the remark this answers is not in the document]
Comment (jeffreyCline) [reply website/docs/r/app_service_active_slot.html.markdown]: I'm not sure, but it is exposed in models.go CsmSlotEntity struct so I exposed it to Terraform.

[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply website/docs/r/app_service_active_slot.html.markdown]: taking a look at the Azure CLI and the Portal - neither of these expose this value; as such I think we can safely default this value to `true` for the moment and not expose it :)

[the remark this answers is not in the document]
Comment (jeffreyCline) [reply website/docs/r/app_service_active_slot.html.markdown]: Ok, sounds good... I make that change.

**Comment by tombuildsstuff** [comment]
Tests pass:

screen shot 2018-02-12 at 18 17 32

**Pull request merged**